src/Axe/Controllers/CompilerController.cs

```diff
@@ -16,23 +16,34 @@
 using Axe.ViewModels.CompilerVm;
 using Microsoft.EntityFrameworkCore;
 using Axe.Managers;
+using Microsoft.AspNetCore.NodeServices;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Authorization;
 
 namespace Axe.Controllers
 {
     public class CompilerController : ControllerExt
     {
         private ICompilerManager compileManager;
+        private ITechnologyManager technologyManager;
+        private INodeServices nodeServices;
 
-        public CompilerController(UserManager<ApplicationUser> userManager, ICompilerManager compileManager, AxeDbContext context)
-            : base(userManager, context)
+        public CompilerController(UserManager<ApplicationUser> userManager,
+            ICompilerManager compileManager,
+            ITechnologyManager technologyManager,
+            INodeServices nodeServices,
```

Comment:
`why INodeServices nodeServices` is here? I don't see usages in controller

Reply:
I forgot to clean it, it isn't used here.

Reply:
fe30448